### PR TITLE
Add support for SNS alarms and dashboard

### PR DESCRIPTION
**Pull request by direnakkoc**
## Description

Add support for SNS alarms and dashboard for the metrics are 

- NumberOfNotificationsFilteredOut-InvalidAttributes
- NumberOfNotificationsFailed

## Motivation and Context

-  **NumberOfNotificationsFilteredOut-InvalidAttributes** - This metric keeps track of messages that were filtered out because they carried invalid or malformed attributes and, thus, didn’t match the subscription filter policy.

- **NumberOfNotificationsFailed** – This last metric tracks all the messages that failed to be delivered to subscribing endpoints, regardless of whether a filter policy had been set for the endpoint. This metric is emitted after the message delivery retry policy is exhausted, and SNS stops attempting to deliver the message. At that moment, the subscribing endpoint is likely no longer reachable. For example, the subscribing SQS queue or Lambda function has been deleted by its owner. You may want to closely monitor this metric to address message delivery issues quickly.

## How Has This Been Tested?

- Unit test added
- Integration test is completed

## Screenshots (if appropriate):
Screenshot 2022-06-21 at 15 28 36
Screenshot 2022-06-21 at 15 31 24

## Types of changes

- [x] New feature (non-breaking change which adds functionality)

## Checklist:

- [x] My code follows the code style of this project.
- [x] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING.md** document.
- [x] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -141,10 +142,16 @@ The dashboard contains one widget per Step Function:
 
 ### ECS / Fargate
 
-ECS alarms are created for Fargate or EC2 clusters:
```
Comment (eoinsha) [inline README.md:144]: Why did we remove "Fargate or EC2 clusters"?

File: serverless-plugin/default-config.yaml [hunk 2]
```diff
@@ -94,6 +94,13 @@ alarms:
     CPUUtilization:
       Statistic: Average
       Threshold: 90
+  SNS:
```
Comment (eoinsha) [inline serverless-plugin/default-config.yaml:97]: The defaults can be added to the configuration section of the project README also

File: serverless-test-project/package.json [hunk 3]
```diff
@@ -19,5 +19,9 @@
     "test": "ALARM_TOPIC=test sls package",
     "test:v2": "ALARM_TOPIC=test sls package -c serverless-v2.yml"
   },
-  "gitHead": "c4300240f9e854c4eb1c5503b839882ff0cd1cae"
+  "gitHead": "c4300240f9e854c4eb1c5503b839882ff0cd1cae",
+  "dependencies": {
+    "https": "^1.0.0",
```
Comment (eoinsha) [inline serverless-test-project/package.json:24]: Node.js has a built-in `https` module, so I don't think we need to add an NPM module. Does it add anything?
I'd like to suggest replacing this with `axios` in any case and changing the handler to use `async`/`await`.

File: serverless-test-project/package.json [hunk 4]
```diff
@@ -19,5 +19,9 @@
...
+    "sns-validator": "^0.3.4"
```
Comment (eoinsha) [inline serverless-test-project/package.json:25]: SNS validation is probably overkill for our test, so I'd suggest to remove this.

**Review by eoinsha — CHANGES_REQUESTED** [review]

File: README.md (hunk 1 above)
Comment (direnakkocdemir) [reply README.md:144]: Sorry, it happened by by accident. I added back.

File: serverless-plugin/default-config.yaml (hunk 2 above)
Comment (direnakkocdemir) [reply serverless-plugin/default-config.yaml:97]: Yes, defaults have been added indeed. Please check default alarms in the line **276** and default dashboard the line **363** in the README

**Pull request merged**